CMakeLists.txt

```diff
@@ -1,76 +1,20 @@
-cmake_minimum_required (VERSION 2.6)
-project (carpet)
+cmake_minimum_required(VERSION 3.2)
+project(carpet)
 set(CMAKE_CXX_STANDARD 14)
 
 ENABLE_TESTING()
 include(CTest)
 INCLUDE(Dart)
 
-
-# The version number.
-set (Tutorial_VERSION_MAJOR 1)
-set (Tutorial_VERSION_MINOR 0)
-
-SET(CMAKE_SYSTEM_NAME Linux)
-SET(CMAKE_C_COMPILER arm-linux-gnueabihf-gcc)
-SET(CMAKE_ASM_COMPILER arm-linux-gnueabihf-gcc)
-SET(CMAKE_SYSTEM_PROCESSOR arm)
-
 find_package(libcluon REQUIRED)
 include_directories(SYSTEM ${CLUON_INCLUDE_DIRS})
 
-## Strict warning level
-IF (CMAKE_BUILD_TOOL MATCHES "(msdev|devenv|nmake)")
-  # Use the highest warning level for visual studio.
-  SET(CMAKE_CXX_WARNING_LEVEL 4)
-  IF (CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
-    STRING(REGEX REPLACE "/W[0-4]" "/W4" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-  ELSE (CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4")
-  ENDIF (CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
-
-  SET(CMAKE_C_WARNING_LEVEL 4)
-  IF (CMAKE_C_FLAGS MATCHES "/W[0-4]")
-    STRING(REGEX REPLACE "/W[0-4]" "/W4" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
-  ELSE (CMAKE_C_FLAGS MATCHES "/W[0-4]")
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /W4")
-  ENDIF (CMAKE_C_FLAGS MATCHES "/W[0-4]")
-
-ELSEIF (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_GNUC)
-  # use -Wall for gcc
-  IF(NOT CMAKE_CXX_FLAGS MATCHES "-Wall$")
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
-  ENDIF(NOT CMAKE_CXX_FLAGS MATCHES "-Wall$")
-
-  IF(NOT CMAKE_C_FLAGS MATCHES "-Wall$")
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
-  ENDIF(NOT CMAKE_C_FLAGS MATCHES "-Wall$")
+add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/messages.cpp
```

Comment:
Maybe we should comment this more for future reference?

Reply:
Referring to the entire section(not the removed parts) as i am not about the entirety of what is happening myself.